src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -35,6 +44,10 @@ def __init__(self, spiffe_socket: str = None) -> None:
                 'SPIFFE socket argument or environment variable invalid in DefaultWorkloadApiClient.'
             )
 
+        self._channel = grpc_client.insecure_channel(
+            spiffe_socket=self._config.spiffe_endpoint_socket
+        )
+
```

Comment:
I'm a little confuse here. 
The way I imagine this is with `DefaultWorkloadApiClient` having a property `grpc_client` that receives the `SpiffeWorkloadAPIStub` already created. So the creation of the channel should be part of the initialization of the `GrpcClient`.

```
class DefaultGrpcClient(GrpcClient):
    def __init__(spiffe_socket: str) -> None:
        channel = grpc_insecure_channel(spiffe_socket)
        return SpiffeWorkloadAPIStub(channel)
```
And then:
```
class DefaultWorkloadApiClient(WorkloadApiClient):
    def __init__(
        self,
        spiffe_socket: str = None,
        grpc_client: GrpcClient = cast(GrpcClient, DefaultGrpcClient),
    ) -> None:
...
    self. grpc_client = grpc_client
...
```

That way we can achieve the goal of using a mock of the GrpcClient for unit testing.

I might be missing something, if that is the case, sorry if this comment make things confusing.

Reply:
Maybe? That's why I published this PR. :-) I'm not sure the best approach myself.

But, at this point I want to see what @maxlambrecht has come up with, which will change all of this anyway, I think.